wrp/header_wrp.go

```diff
@@ -0,0 +1,150 @@
+package wrp
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// Constant HTTP header strings representing WRP fields
+const (
+	MsgTypeHeader        = "X-Midt-Msg-Type"
+	ContentTypeHeader    = "X-Midt-Content-Type"
+	AcceptHeader         = "X-Midt-Accept"
+	TrasactionUuidHeader = "X-Midt-Transaction-Uuid"
+	StatusHeader         = "X-Midt-Status"
+	RDRHeader            = "X-Midt-Request-Delivery-Reponse"
+	HeadersArrHeader     = "X-Midt-Headers"
+	IncludeSpansHeader   = "X-Midt-Include-Spans"
+	SpansHeader          = "X-Midt-Spans"
+	PathHeader           = "X-Midt-Path"
+	SourceHeader         = "X-Midt-Source"
+)
+
+// Map string to MessageType int
+func StringToMessageType(str string) MessageType {
+	switch str {
+	case "Auth":
+		return AuthMessageType
+	case "SimpleRequestResponse":
+		return SimpleRequestResponseMessageType
+	case "SimpleEvent":
+		return SimpleEventMessageType
+	case "Create":
+		return CreateMessageType
+	case "Retrieve":
+		return RetrieveMessageType
+	case "Update":
+		return UpdateMessageType
+	case "Delete":
+		return DeleteMessageType
+	case "ServiceRegistration":
+		return ServiceRegistrationMessageType
+	case "ServiceAlive":
+		return ServiceAliveMessageType
+	default:
+		return -1
+	}
+}
+
+// Convert HTTP header to WRP generic Message
+func HeaderToWRP(header http.Header) (*Message, error) {
+	msg := new(Message)
+
+	// MessageType is mandatory
+	msgType := header.Get(MsgTypeHeader)
+	if !strings.EqualFold(msgType, "") && StringToMessageType(msgType) != MessageType(-1) {
```

Comment:
You can cut down on the string match by assigning the output of `StringToMessageType()` to a variable before this if statement.  Then just check all at once & possibly fail.  It will slow down the failure cases, but speed up the normal case.